tests/Cache/FileCacheTest.php

```diff
@@ -92,16 +93,48 @@ public function testMissingCacheFileRead(): void
         $cache->read('missing_file');
     }
 
-    public function testUnwriteableCacheDir(): void
+    public function testUnwritableCachePath(): void
+    {
+        $root = vfsStream::setup();
+        $root->addChild(new vfsStreamDirectory($this->getCacheDirName(), 0));
+
+        $this->expectExceptionMessageMatches('/Cache path "[^"]+" is not writeable/');
+        $this->expectException(CacheException::class);
+
+        new FileCache($root->url());
+    }
+
+    public function testUncreatableCachePath(): void
     {
         $root = vfsStream::setup();
         $root->chmod(0);
 
+        $this->expectExceptionMessageMatches('/Cache path "[^"]+" cannot be created or is not a directory: .+ cannot be created\./');
         $this->expectException(CacheException::class);
 
         new FileCache($root->url());
     }
 
+    public function testNonDirectoryCachePath(): void
+    {
+        $root = vfsStream::setup();
+        $root->addChild(new class($this->getCacheDirName(), 0) extends vfsStreamDirectory {
+            public function appliesTo($name)
+            {
+                trigger_error('Some warning', E_USER_WARNING);
```

Comment:
I'm not a fan of this, but I couldn't find any other way to make `mkdir()` trigger an error "natively".